cnd/src/storage/db/commands.rs

```diff
@@ -117,6 +130,35 @@ pub fn update_order_of_swap_to_failed(conn: &SqliteConnection, swap_id: LocalSwa
     Ok(())
 }
 
+pub fn update_order_of_swap_to_settling(
+    conn: &SqliteConnection,
+    swap_id: LocalSwapId,
+) -> Result<()> {
+    let (order, btc_dai_order) = orders::table
+        .inner_join(order_swaps::table.inner_join(swaps::table))
+        .inner_join(btc_dai_orders::table)
+        .filter(swaps::local_swap_id.eq(Text(swap_id)))
+        .select((orders::all_columns, btc_dai_orders::all_columns))
+        .first::<(Order, BtcDaiOrder)>(conn)
+        .with_context(|| NoOrderForSwap(swap_id))?;
+
+    let affected_rows = diesel::update(&btc_dai_order)
```

Comment:
I think we should add a `warn` if we move a previously failed swap to settling.

Reply:
Why `warn`? I feel it is expected behaviour for swaps to be respawned after they are failed and here we are just making sure the accounting is correct :)